pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler_test.go

```diff
@@ -0,0 +1,586 @@
+					if container.Name == constants.InferenceServiceContainerName {
+						if container.Resources.Limits == nil {
+							container.Resources.Limits = make(map[corev1.ResourceName]resource.Quantity)
+						}
+						if container.Resources.Requests == nil {
+							container.Resources.Requests = make(map[corev1.ResourceName]resource.Quantity)
+						}
+						container.Resources.Requests[constants.NvidiaGPUResourceType] = resource.MustParse("5")
+						container.Resources.Limits[constants.NvidiaGPUResourceType] = resource.MustParse("5")
+					}
+				}
+				return updatedArgs
+			},
+			modifyExpected: func(updatedExpected []*appsv1.Deployment) []*appsv1.Deployment {
+				// Overwrite the environment variable
+				for j, envVar := range updatedExpected[0].Spec.Template.Spec.Containers[0].Env {
+					if envVar.Name == constants.TensorParallelSizeEnvName {
+						updatedExpected[0].Spec.Template.Spec.Containers[0].Env[j].Value = "1"
+						break
+					}
+				}
+				for _, deploy := range updatedExpected {
+					deploy.Spec.Template.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							constants.NvidiaGPUResourceType: resource.MustParse("1"),
+						},
+						Limits: corev1.ResourceList{
+							constants.NvidiaGPUResourceType: resource.MustParse("1"),
+						},
+					}
+				}
+
+				return updatedExpected
+			},
+		},
+	}
+
+	for _, tt := range podSpec_tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// retrieve args, expected
+			ttArgs := getDefaultArgs()
+			ttExpected := getDefaultExpectedDeployment()
+
+			// update objectMeta using modify func
+			got := createRawDeployment(ttArgs.objectMeta, ttArgs.workerObjectMeta, ttArgs.componentExt, tt.modifyArgs(ttArgs).podSpec, ttArgs.workerPodSpec)
+
+			// update expected value using modifyExpected func
+			expected := tt.modifyExpected(ttExpected)
+
+			for i, deploy := range got {
+				if diff := cmp.Diff(expected[i], deploy, cmpopts.IgnoreFields(appsv1.Deployment{}, "Spec.Template.Spec.SecurityContext"),
+					cmpopts.IgnoreFields(appsv1.Deployment{}, "Spec.Template.Spec.RestartPolicy"),
+					cmpopts.IgnoreFields(appsv1.Deployment{}, "Spec.Template.Spec.TerminationGracePeriodSeconds"),
+					cmpopts.IgnoreFields(appsv1.Deployment{}, "Spec.Template.Spec.DNSPolicy"),
+					cmpopts.IgnoreFields(appsv1.Deployment{}, "Spec.Template.Spec.AutomountServiceAccountToken"),
+					cmpopts.IgnoreFields(appsv1.Deployment{}, "Spec.Template.Spec.SchedulerName"),
+					cmpopts.IgnoreFields(appsv1.Deployment{}, "Spec.Strategy.Type"),
+					cmpopts.IgnoreFields(appsv1.Deployment{}, "Spec.Strategy.RollingUpdate"),
+					cmpopts.IgnoreFields(appsv1.Deployment{}, "Spec.RevisionHistoryLimit"),
+					cmpopts.IgnoreFields(appsv1.Deployment{}, "Spec.ProgressDeadlineSeconds")); diff != "" {
+					t.Errorf("Test %q unexpected deployment (-want +got): %v", tt.name, diff)
+				}
+			}
+		})
+	}
+}
+
+func int32Ptr(i int32) *int32 {
```

Comment:
instead of adding these you can do `unsafe.Pointer(&b)`

Reply:
Is there a specific reason that I should replace this with your suggestion?

Reply:
these are native from golang. no need to add extra methods.

Reply:
So, you want to eliminate the function and use unsafe.Pointer(&b), correct? I'm not quite sure how to simplify it using unsafe.Pointer(&b) instead of the function. Could you please provide some recommendations?